### PR TITLE
Refactor docker_build rules to use go_image

### PR DESCRIPTION
**Pull request by fejta**
/assign @mattmoor @stevekuznetsov @cjwagner 

Follow-up to https://github.com/kubernetes/test-infra/pull/5620

### CODE REVIEW

**Comment by cjwagner** [comment]
This LGTM, but @mattmoor should also take a look.

**Comment by fejta** [comment]
I don't really think that is necessary

File: experiment/commenter/BUILD [hunk 1]
```diff
@@ -39,8 +26,9 @@ load(
     "go_test",
 )
 
-go_binary(
-    name = "commenter",
+go_image(
+    name = "commenter-image",
+    base = "@distroless-base//image",
```
Comment (mattmoor) [inline experiment/commenter/BUILD]: This should be unnecessary

File: experiment/commenter/BUILD [hunk 2]
```diff
@@ -72,3 +60,10 @@ go_test(
     importpath = "k8s.io/test-infra/experiment/commenter",
     deps = ["//prow/github:go_default_library"],
 )
+
+# Delete after gazelle stops forcing its creation
```
Comment (mattmoor) [inline experiment/commenter/BUILD]: Alternately I (bregrudgingly) added another form of this:
```python
go_image(
   name = "foo",
   binary = "bar",
)
```

For `lang_image` that don't benefit from layering, this is no worse and sidesteps this issue.

File: robots/issue-creator/BUILD [hunk 3]
```diff
@@ -46,8 +29,9 @@ go_library(
     ],
 )
 
-go_binary(
-    name = "issue-creator",
+go_image(
+    name = "issue-creator-image",
+    base = "@distroless-base//image",
```
Comment (mattmoor) [inline robots/issue-creator/BUILD]: unnecessary

**Pull request merged**